tracing/src/main/java/io/airlift/tracing/OpenTelemetryModule.java

```diff
@@ -0,0 +1,113 @@
+package io.airlift.tracing;
+
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import io.airlift.node.NodeInfo;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+
+import static com.google.common.base.StandardSystemProperty.JAVA_VM_NAME;
+import static com.google.common.base.StandardSystemProperty.JAVA_VM_VENDOR;
+import static com.google.common.base.StandardSystemProperty.JAVA_VM_VERSION;
+import static com.google.common.base.StandardSystemProperty.OS_ARCH;
+import static com.google.common.base.StandardSystemProperty.OS_NAME;
+import static com.google.common.base.StandardSystemProperty.OS_VERSION;
+import static com.google.common.base.Strings.nullToEmpty;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.airlift.tracing.Tracing.attribute;
+import static java.util.Objects.requireNonNull;
+
+public class OpenTelemetryModule
+        implements Module
+{
+    private final String serviceName;
+    private final String serviceVersion;
+
+    public OpenTelemetryModule(String serviceName, String serviceVersion)
```

Comment:
Could we infer this from environment variables via launcher.py or via the jar manifest? It looks like we set some properties at build in the jar manifest like `artifactId` and `version`

Reply:
We decided it's better to have people set this directly, since the name and version comes from the application.